zebra-consensus/src/block.rs

```diff
@@ -53,8 +53,15 @@ pub(crate) fn node_time_check(
     }
 }
 
+/// Check that there is exactly one coinbase transaction in `Block`, and that
+/// the coinbase transaction is the first transaction in the block.
+///
+/// "The first (and only the first) transaction in a block is a coinbase
+/// transaction, which collects and spends any miner subsidy and transaction
+/// fees paid by transactions included in this block."[S 3.10][3.10]
+///
 /// [3.10]: https://zips.z.cash/protocol/protocol.pdf#coinbasetransactions
-pub(crate) fn coinbase_check(block: &Block) -> Result<(), Error> {
+pub(crate) fn coinbase_is_first_check(block: Arc<Block>) -> Result<(), Error> {
```

Comment:
I also think we might want to make this to be a method on `Block`

Reply:
Moved the coinbase check in ae5d88c60fb2a7da5ca6743f8440f8247117e78a.

Also moved the time check in a8813ff0044c1fc08a016d0234aaa4536ed78c08.